examples/go/stream_benchmarker/benchmark/main.go

```diff
@@ -0,0 +1,261 @@
+	b.WriteString("Total Bytes\t" + strconv.FormatInt(sessionSummary.totalBytes, 10) + "\n")
+	b.WriteString("Average Speed\t" + fmt.Sprintf("%f", sessionSummary.totalMbps/sessionSummary.numIntervals) + "\n")
+	b.WriteString("Total Out Of Order Bytes\t" + strconv.FormatInt(sessionSummary.totalOutOfOrderData, 10) + "\n")
+
+	fmt.Println(b.String())
+}
+
+func main() {
+
+	apiKey := flag.String("key", "colin-alpha-key.json", "api key")
+	endpoint := flag.String("endpoint", "api-alpha.stellarstation.com:443", "api endpoint")
+	satelliteID := flag.String("id", "112", "satellite id")
+	interval := flag.Int("interval", 10, "reporting interval")
+	exitOnNoDataInSeconds := flag.Int("exitOnNoDataIn", -1, "number of seconds of no data received to auto exit program")
+	printSessionSummary := flag.Bool("summary", true, "print a session summary")
+
+	flag.Parse()
+
+	satelliteStreamOptions := &benchmark.SatelliteStreamOptions{
+		AcceptedFraming: []stellarstation.Framing{stellarstation.Framing_AX25, stellarstation.Framing_BITSTREAM},
+		SatelliteID:     *satelliteID,
+		APIKey:          *apiKey,
+		Endpoint:        *endpoint,
+	}
+
+	var streamChannel = make(chan *stellarstation.Telemetry)
+
+	ss, err := benchmark.OpenSatelliteStream(satelliteStreamOptions, streamChannel)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	ticker := time.NewTicker(time.Second * time.Duration(*interval))
+	defer ticker.Stop()
+
+	fmt.Println("Listening for messages")
+	var sessionSummary sessionSummary
+	var md metricsData
+	var storeMostRecentTimeLastByteReceived time.Time = time.Time{}
+	var gotFirstMessage = false
+	var done = false
+	for !done {
+		select {
+		case message := <-streamChannel:
+			if gotFirstMessage == false {
+				fmt.Println("Receiving messages")
+				printHeader()
+				gotFirstMessage = true
+				firstTime, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+				md.initialTime = firstTime
+				sessionSummary.firstByteTime = firstTime
+			}
+			timeFirstByteReceived, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+			timeLastByteReceived, _ := ptypes.Timestamp(message.TimeLastByteReceived)
+			numBytesInMessage := len(message.Data)
+
+			metric := metric{
+				timeFirstByteReceived: timeFirstByteReceived,
+				timeLastByteReceived:  timeLastByteReceived,
+				dataSize:              numBytesInMessage,
+				metricTime:            time.Now().UTC(),
+			}
+			md.metrics = append(md.metrics, metric)
+			storeMostRecentTimeLastByteReceived = timeLastByteReceived
+			md.mostRecentTimeLastByteReceived = storeMostRecentTimeLastByteReceived
+		case timeOfReporting := <-ticker.C:
+			if gotFirstMessage {
+				md.timeOfReporting = timeOfReporting
+				outputDetails := md.compileDetails()
+				sessionSummary.totalBytes += outputDetails.totalDataSize
```

Comment:
This can be made into a method on the session summary type that takes an output details type as input.